### ISSUE TITLE
Starting a process instance by a message event doesn't store the business key

### ISSUE THREAD

**Issue by Adam-Higginson** [Activiti/Activiti#2270]
If you attempt to start a process instance by a message you can pass in a business key as a parameter to store against the process instance. An example of doing this would be:

```
processEngine.getRuntimeService()
			.startProcessInstanceByMessage(message, 
                                                        someBusinessKey,
							processParameters);
```

The business key given gets correctly passed through to the `StartProcessInstanceByMessageCmd` instance and is stored in a field called `businessKey`, however it is then never used. When the `createAndStartProcessInstanceByMessage` method is called in `ProcessInstanceHelper` it doesn't accept a business key as an argument and instead passes null through to the `createAndStartProcessInstanceWithInitialFlowElement` method. This causes a null business key to then be stored against the process.

**Comment by salaboy** [comment]
@Adam-Higginson are you using Activiti 6.x?

**Comment by Adam-Higginson** [comment]
Yep, I noticed this when trying to update an Activiti 5 setup to Activiti 6

**Comment by salaboy** [comment]
@Adam-Higginson ok.. so we are working full speed on 7 which is built on top of 6, but we are not covering messages just yet. That will come in a not distant future. We will probably try to fix it in that the 7.x series.
